src/probe_modules/module_dns.c

```diff
@@ -1083,11 +1084,48 @@ probe_module_t module_dns = {
 	"This module sends out DNS queries and parses basic responses. "
 	"By default, the module will perform an A record lookup for "
 	"google.com. You can specify other queries using the --probe-args "
-	"argument in the form: 'type,query', e.g. 'A,google.com'. The module "
+	"argument in the form: 'type,query', e.g. 'A,google.com'. The --probes/-P "
+	"flag must be set to a multiple of the number of DNS questions. The module "
 	"supports sending the the following types: of queries: A, NS, CNAME, SOA, "
 	"PTR, MX, TXT, AAAA, RRSIG, and ALL. The module will accept and attempt "
 	"to parse all DNS responses. There is currently support for parsing out "
 	"full data from A, NS, CNAME, MX, TXT, and AAAA. Any other types will be "
 	"output in raw form."
 
 };
+
+// find_num_dns_questions parses the --probe-args supplied by the user and finds how many DNS questions they want to probe
+// Sets the global static num_questions
+static void set_num_dns_questions(struct state_conf *conf) {
+	// default number of questions is 1, if the user doesn't input any probe_args
+	num_questions = 1;
+	if (!conf->probe_args) {
+		// user didn't input any probe args, we'll default to google.com
+		return;
+	}
+
+	// if a user inputs a single query "AAAA,google.com;", the semicolon will break
+	// the question-counting logic below. Strip any leading/trailing semicolons
+	if (*conf->probe_args == probe_arg_delimitor && strlen(conf->probe_args) == 1) {
+		// user only input ";" as probe-args, error immediately
+		log_fatal("dns",
+			  "Invalid probe args (%s). Format: \"A,google.com\" or \"A,google.com;A,example.com\"", conf->probe_args);
+	}
+	if (*conf->probe_args == probe_arg_delimitor) {
+		log_debug("dns", "Probe args (%s) contains leading semicolon. Stripping.", conf->probe_args);
+		conf->probe_args = conf->probe_args + 1;
+	}
+	if (conf->probe_args[strlen(conf->probe_args) - 1] == probe_arg_delimitor) {
+		log_debug("dns", "Probe args (%s) contains trailing semicolon. Stripping.", conf->probe_args);
+		conf->probe_args[strlen(conf->probe_args) - 1] = '\n';
+	}
+	// find how many probe_args the user wants to query
+	if (conf->probe_args) {
+		int arg_strlen = strlen(conf->probe_args);
+		for (int i = 0; i < arg_strlen; i++) {
```

Comment:
I think that the more canonical way to do this would be to use `strtok` or `strtok`. More broadly though, I do wonder if it makes sense that this is logic is different from the logic we use for actually _parsing out_ the questions. It seems that when parsing, we would know this information already and then wouldn't have this logic in different places.

Reply:
Fair, and thanks for the suggestion to look at `stroke`, seems very much like the correct way to go about this. I've re-done the logic and it seems much cleaner. I've re-tested the scenarios I outlined in the PR, mind taking a second look and we can merge if it looks good to you? @zakird

Reply:
Update: let me integrate the changes made [here](https://github.com/zmap/zmap/pull/764), they'll conflict.

Reply:
I'll re-assign you when done.